File: src/pymax/types/domain/name.py
from .base import CamelModel


class Name(CamelModel):
    """Имя пользователя.

    :ivar name: Полное отображаемое имя.
    :vartype name: str | None
    :ivar first_name: Имя.
    :vartype first_name: str | None
    :ivar last_name: Фамилия.
    :vartype last_name: str | None
    :ivar type: Тип имени.
    :vartype type: str | None
    """

    name: str | None = None
    first_name: str | None = None
    last_name: str | None = None
    type: str | None = None

File: src/PHPMax/Domain/Name.php
<?php

declare(strict_types=1);

namespace PHPMax\Domain;

use PHPMax\Support\Model;

class Name extends Model
{
    /** @var string|null */
    public $name;
    /** @var string|null */
    public $firstName;
    /** @var string|null */
    public $lastName;
    /** @var string|null */
    public $type;

    protected static function schema(): array
    {
        return [
            'name' => ['type' => 'string'],
            'firstName' => ['type' => 'string'],
            'lastName' => ['type' => 'string'],
            'type' => ['type' => 'string'],
        ];
    }
}
